client-misk-hibernate/src/main/kotlin/app/cash/backfila/client/misk/hibernate/internal/HibernateBackfillOperator.kt

```diff
@@ -196,8 +191,12 @@ internal class HibernateBackfillOperator<E : DbEntity<E>, Pkey : Any, Param : An
         val stopwatch = Stopwatch.createStarted()
         boundingMax = boundingRangeStrategy
           .computeBoundingRangeMax(
-            backfill, partitionName, previousEndKey, backfillRange,
-            scanSize
+            backfill,
+            partitionName,
+            previousEndKey,
+            backfillRange.start.let { primaryKeyCursorMapper.fromByteString(backfill.pkeyClass.java, it).getOrThrow() },
```

Comment:
maybe the class should expose a conversion method instead. I'd think the mapper and the backfill should continue to be related.

Reply:
The type is independent of the backfill, just like how it's bound in hibernate independent of the backfill